tera-templates/macros.html

```diff
@@ -0,0 +1,77 @@
+{# Constructs a title based on the given crate name and version #}
+{% macro doc_title(name, version) %}
+    {%- if name -%}
+        {{ name }} {{ version | default(value="") }} - Docs.rs
+    {%- else -%}
+        Docs.rs
+    {%- endif -%}
+{% endmacro doc_title %}
+
+{#
+    Makes the appropriate JS imports for highlighting
+    * `languages` An array of strings where each is a valid highlight.js language
+#}
+{% macro highlight_js(languages) %}
+    {# Load the highlight script #}
+    <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.4.0/highlight.min.js" type="text/javascript"
+        charset="utf-8"></script>
+
+    {# Load the script for each provided language #}
+    {%- for language in languages -%}
+    <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.4.0/languages/{{ language }}.min.js"
+        type="text/javascript" charset="utf-8"></script>
+    {%- endfor -%}
+
+    {# Activate highlighting #}
+    <script type="text/javascript" charset="utf-8">
+        hljs.initHighlighting();
+    </script>
+{% endmacro highlight_js %}
+
+{# Makes the appropriate CSS imports for highlighting #}
+{% macro highlight_css() %}
+    {# Load the highlighting theme css #}
+    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.4.0/styles/github.min.css"
+        type="text/css" media="all" />
+{% endmacro highlight_css %}
```

Comment:
Is this used in every page? If so, we should consider making it part of the default `{% block css %}` instead having it as a macro.

Reply:
It's only used on some pages, hence the macro